client/jetpack-connect/index.js

```diff
@@ -26,15 +27,26 @@ export default function() {
 		clientRender
 	);
 
-	page(
-		'/jetpack/connect/:type(install)/:locale?',
-		controller.redirectWithoutLocaleIfLoggedIn,
-		controller.persistMobileAppFlow,
-		controller.setMasterbar,
-		controller.connect,
-		makeLayout,
-		clientRender
-	);
+	if ( config.isEnabled( 'jetpack/connect/install' ) ) {
+		page(
+			'/jetpack/connect/install',
+			controller.persistMobileAppFlow,
```

Comment:
I don't think we expect mobile apps to arrive at this route, so we can leave out this `persistMobileAppFlow` function.